js/core/config.js

```diff
@@ -44,13 +44,25 @@ const config = {
         if(optionsString.trim().charAt(0) !== '{') {
             optionsString = '{' + optionsString + '}';
         }
+
         try {
-            // eslint-disable-next-line no-new-func
-            return (new Function('return ' + optionsString))();
+            return JSON.parse(optionsString);
         } catch(ex) {
-            throw errors.Error('E3018', ex, optionsString);
+            try {
+                return JSON.parse(normalizeToJSONString(optionsString));
+            } catch(exNormalize) {
+                throw errors.Error('E3018', ex, optionsString);
+            }
         }
-    }
+    },
+};
+
+const normalizeToJSONString = (optionsString) => {
+    return optionsString
+        .replace(/([{,])[\s\n]*'([^']+)'[\s\n]*:/g, '$1"$2":') // replace quotes for single quoted keys
```

Comment:
Isn't it better to replace all single quotes first (like we do it on the line 64) and then do not care about keys wrapped by single quotes?

Reply:
reasonable..